tools/autograd/templates/variable_factories.h

```diff
@@ -186,12 +186,23 @@ AT_FORALL_SCALAR_TYPES_AND3(Bool, Half, BFloat16, TENSOR)
 /// support it in the future by iterating over all sub-lists to find
 /// the largest data type that can represent all of the elements, or by using
 /// variadic templates.
-inline at::Tensor tensor(detail::InitListTensor list_init_tensor, const at::TensorOptions& options) {
-  return autograd::make_variable(list_init_tensor.to_tensor(options), options.requires_grad());
+inline at::Tensor tensor(detail::InitListTensor init_list_tensor, const at::TensorOptions& options) {
+  return autograd::make_variable(init_list_tensor.to_tensor(options), options.requires_grad());
 }
 
-inline at::Tensor tensor(detail::InitListTensor list_init_tensor) {
-  return torch::tensor(list_init_tensor, at::dtype(list_init_tensor.scalar_type()));
+inline at::Tensor tensor(detail::InitListTensor init_list_tensor) {
+  return torch::tensor(init_list_tensor, at::dtype(init_list_tensor.scalar_type()));
+}
+
+/// NOTE: We add `torch::tensor(std::initializer_list<detail::InitListTensor>)` function overload,
+/// so that `torch::tensor({{1, 2}})` can take this overload instead of `torch::tensor(at::ArrayRef<T>)`.
+inline at::Tensor tensor(std::initializer_list<detail::InitListTensor> init_list) {
```

Comment:
Do we need another override for `inline at::Tensor tensor(std::initializer_list<detail::InitListTensor> init_list, TensorOptions)`? (And also a test)